docs/installing-open-mpi/packagers.rst

```diff
@@ -80,15 +80,59 @@ running Open MPI's ``configure`` script.
 
 .. _label-install-packagers-dso-or-not:
 
-Components ("plugins"): DSO or no?
-----------------------------------
+Components ("plugins"): static or DSO?
+--------------------------------------
 
 Open MPI contains a large number of components (sometimes called
 "plugins") to effect different types of functionality in MPI.  For
 example, some components effect Open MPI's networking functionality:
 they may link against specialized libraries to provide
 highly-optimized network access.
 
+Open MPI can build its components as Dynamic Shared Objects (DSOs) or
+statically included in core libraries (regardless of whether those
+libraries are built as shared or static libraries).
+
+.. note:: As of Open MPI |ompi_ver|, ``configure``'s global default is
+          to build all components as static (i.e., part of the Open
+          MPI core libraries, not as DSOs).  Prior to Open MPI v5.0.0,
+          the global default behavior was to build most components as
+          DSOs.
+
+Why build components as DSOs?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are advantages to building components as DSOs:
+
+* Open MPI's core libraries |mdash| and therefore MPI applications
+  |mdash| will have very few dependencies.  For example, if you build
+  Open MPI with support for a specific network stack, the libraries in
+  that network stack will be dependencies of the DSOs, not Open MPI's
+  core libraries (or MPI applications).
+
+* Removing Open MPI functionality that you do not want is as simple as
+  removing a DSO from ``$libdir/open-mpi``.
+
+Why build components as part of Open MPI's core libraries?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The biggest advantage to building the components as part of Open MPI's
+core libraries is when running at (very) large scales when Open MPI is
+installed on a network filesystem (vs. being installed on a local
+filesystem).
+
+Specifically, when running thousands or MPI processes (or more) that
+each query and open dozens of DSO files on a network filesystem during
+:ref:``MPI_Init(3) <mpi_init>``, this can lead to a bottleneck of
+network performance.
+
+In this scenario, it is better to compile all the components
+statically as part of the core libraries such that this kind of
```

Comment:
I would be pedantic here and replace "statically" with something that doesn't conjure .a files.  It's unfortunate the overlap in terms between DSO and SO make this verbiage hard.

Reply:
Talked with @lrbison out-of-band and agreed in some revamped text for this section.  See new push.